### DISCUSSION TITLE
I want transparency for applications window

### DISCUSSION THREAD

**Discussion by smjt2000** [qtile/qtile#4216] [Q&A]
Hi
I want to transparent all windows except someones and I don't know how to do that.
I tried this as I see [here](https://docs.qtile.org/en/latest/manual/commands/api/windows.html#libqtile.backend.base.Window.set_opacity) but it doesn't work
```python
@hook.subscribe.client_new
def set_transparency(window):
    name = window.name
    window.set_opacity(0.8)
```
the `name` is used to do some exception like this:
```python
if name.find("Konsole") == -1:  # this mean if the window is not konsole
    # set opacity here
```

is there anyway to do that??

**Comment by Sydiepus** [comment]
If you're using X11 you can use a compositor like [picom](https://github.com/yshui/picom) that will let you change windows opacity and add exceptions.

**Reply by elParaguayo** [reply]
If they're on x11, transparency won't work at all without picom.

Also, the `set_opacity` command is only available in latest git version. It's not in the 0.22.1 release, they'd need `window.opacity = 0.8` in that scenario.

**Comment by rmcmilli** [comment]
How would one get transparency in wayland?

**Reply by 0xUnicorn** [reply]
> How would one get transparency in wayland?

I used this hook to trigger a function validating the window name before setting the opacity

```python
from libqtile.config import Match
from libqtile import hook

@hook.subscribe.client_new
def set_opacity(window):
    if window.match(Match(wm_class="Alacritty")):
        window.opacity = 0.8
```
